js/src/forum/components/ReactionsModal.js

```diff
@@ -0,0 +1,101 @@
+import Modal from 'flarum/components/Modal';
+import avatar from 'flarum/helpers/avatar';
+import username from 'flarum/helpers/username';
+import emoji from '../../common/util/emoji';
+
+
+export default class ReactionsModal extends Modal {
+    className() {
+        return 'ReactionsModal Modal--small';
+    }
+
+    title() {
+        return app.translator.trans('fof-reactions.forum.modal.title');
+    }
+
+    init() {
+        this.idToReaction = app.forum.reactions()
+        .filter(x => x['data']['attributes']['display'])
+        .reduce((a, x) => ({...a, [Number(x['data']['id'])]: x['data']['attributes']['display']}), {});
+
+        this.groupedReactions = this.groupReactions(this.props.post.reactions());
+        this.groupedReactions.forEach(g => {
+            g.forEach(r => {
+                app.store.find('users', r['userId']).then(user => {
+                    r['modalUser'].push(
+                        <li>
+                            <a className="ModalUser" href={app.route.user(user)} config={m.route}>
+                                {avatar(user)} {username(user)}
+                            </a>
+                        </li>
+                    );
+
+                    m.redraw();
+                })
+            })
+        });
+    }
+
+    
+    content() {
+        return (
+            <div className="Modal-body">
+                <ul className="ReactionsModal-list">
+                    {
+                        this.groupedReactions.map(g => {
+                            return (
+                                <div className="ReactionGroup">
+                                <legend>{g[0]['legend']}</legend>
+                                <hr/>
+                                {
+                                    g.map(r =>  r['modalUser'])
+                                }
+                                </div>
+                            )
+                        })
+                    }
+                </ul>
+            </div>
+        );
+    }
+
+    groupReactions(reactions) {
+        // Recreate reactions to only have useful data
+        const informativeReactions = reactions.map(x => {
+            let toReturn = {};
+    
+            toReturn['id'] = Number(x['data']['id'].split('-')[0])
```

Comment:
`reactions` here is an array of Reaction model instances. To access the attributes, all you need to do is `x.id()`, `x.type()`, `x.identifier()`, etc....  Also - could you explain what this code does? What is `legend`? modalUser? modalLegend?

Reply:
Will do, give me an hour or so, I have to check for some possible CSS issues (as the gamification implementation assumes there won't be such a scenario. I'll fix the things you mentioned (I didn't know about the model instance as I only looked at the console log of those variables) as well.